### PR TITLE
[DOC] More on links 

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Replaces https://github.com/ruby/ruby/pull/9011, which I fouled up.

### CODE REVIEW

**Comment by BurdetteLamar** [comment]
I have changed "Use the rdoc-ref" to "Prefer the rdoc-ref."

File: doc/contributing/documentation_guide.md [hunk 1]
```diff
@@ -199,6 +199,36 @@ will be autolinked:
...
+### Explicit Links
+
+When writing an explicit link, follow these guidelines.
+
+#### +rdoc-ref+ Scheme
+
+Prefer the +rdoc-ref+ scheme for:
+
+- A link in core documentation to other core documentation.
+- A link in core documentation to documentation in a standard library package.
```
Comment (jeremyevans) [inline doc/contributing/documentation_guide.md:211]: My feelings have not changed.  We should not recommend an approach unless we can figure out how to make it work ourselves.

One issue with using rdoc-ref in this case is if you build just the core documentation without the standard library documentation, the rdoc-ref link will not work.  I don't think that affects docs.ruby-lang.org.  It looks like for recent releases, ruby-doc.org also builds them both together.  So I don't think we should avoid this recommendation, but we must figure out how to get it to work first.  I'll see if I can determine how to get it to work.

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:211]: I'll work on it, too.

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:211]: Got it.  I'll push soon.

Comment (BurdetteLamar) [reply doc/contributing/documentation_guide.md:211]: Fixed.

**Review by jeremyevans — APPROVED** [review]
Looks good.  Please remember to squash when merging.

**Comment by BurdetteLamar** [comment]
> Looks good. Please remember to squash when merging.

Always hate to merge.  @jeremyevans, will you do it?

**Pull request merged**